webapp/src/components/sidebar_right/github_items.jsx

```diff
@@ -125,4 +158,10 @@ const notificationReasons = {
     team_mention:	'You were on a team that was mentioned.',
 };
 
-export default GithubItems;
+function areEquals(prevProps, nextProps) {
+    return prevProps.theme === nextProps.theme &&
+        prevProps.rhsState === nextProps.rhsState &&
+        prevProps.items.reduce((acc, i) => `${acc}-${i.id}`, '') === nextProps.items.reduce((acc, i) => `${acc}-${i.id}`, '');
+}
+
+export default React.memo(GithubItems, areEquals);
```

Comment:
Nice! 👌

Reply:
Hmm, `React.memo` is a strategy to boost performance by reducing unnecessary renders, but it seems were relying on it here to help enforce business logic about when to transition "old state" so as to achieve the desired rendering logic.

This approach is still buggy. As an example, if I change my theme, the dimmed items will disappear, and it's really hard to fix this.

I'd like to suggest we move all of this logic into a reducer instead. It should be the reducer's responsibility to decide, based on the incoming actions, when to clear the "old state" and when to preserve it.